packages/dds/tree/src/core/checkout/branch.ts

```diff
@@ -0,0 +1,155 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { EventEmitter } from "../../events";
+import { findAncestor, GraphCommit, mintCommit, mintRevisionTag, Rebaser } from "../rebase";
+
+/**
+ * The events emitted by a `SharedTreeBranch`
+ */
+export interface SharedTreeBranchEvents<TChange> {
+	/**
+	 * Fired any time the branch has a new change applied to it
+	 */
+	onChange(change: TChange): void;
+}
+
+/**
+ * A branch of changes that can be applied to a SharedTree.
+ */
+export class SharedTreeBranch<TChange> extends EventEmitter<SharedTreeBranchEvents<TChange>> {
+	private head: GraphCommit<TChange>;
+	private readonly forks = new Set<SharedTreeBranch<TChange>>();
+	private disposed = false;
+
+	/**
+	 * @param getBaseBranch - a function which retrieves the head of the base branch
+	 * @param mergeIntoBase - a function which describes how to merge this branch into its base branch
```

Comment:
What does the return value represent? Does this function mutate the base branch or just return a change which somehow describes how do the merge?

Reply:
Updated docs in an attempt to make this clearer, although hopefully we won't need to implement it any more times.